keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -538,12 +538,12 @@ public DoubleTensor minusInPlace(DoubleTensor that) {
         } else if (this.isScalar()) {
             return this.minus(that);
         } else {
-            INDArrayShim.subi(tensor, unsafeGetNd4J(that), tensor);
+            return new Nd4jDoubleTensor(INDArrayShim.subi(tensor, unsafeGetNd4J(that)));
```

Comment:
Do we *always* want to return the new Tensor?  This seems to undo the point of the inPlace stuff - can we do a check that the value we get back is the same one as before and return a new Tensor only if the array objects differ?

Reply:
Nice, I've implemented a check to see if the tensor returned equals the tensor we're operating on. If they're the same then we just return `this`